zerver/lib/events.py

```diff
@@ -74,62 +74,63 @@ def get_custom_profile_field_values(realm_id: int) -> Dict[int, Dict[str, Any]]:
             }
     return profiles_by_user_id
 
-
 def get_raw_user_data(realm: Realm, user_profile: UserProfile, client_gravatar: bool,
                       include_custom_profile_fields: bool=True) -> Dict[int, Dict[str, str]]:
     user_dicts = get_realm_user_dicts(realm.id)
 
-    if include_custom_profile_fields:
-        profiles_by_user_id = get_custom_profile_field_values(realm.id)
-
-    def user_data(row: Dict[str, Any]) -> Dict[str, Any]:
-        avatar_url = get_avatar_field(
-            user_id=row['id'],
-            realm_id=realm.id,
-            email=row['delivery_email'],
-            avatar_source=row['avatar_source'],
-            avatar_version=row['avatar_version'],
-            medium=False,
-            client_gravatar=client_gravatar,
-        )
-
-        is_admin = row['role'] == UserProfile.ROLE_REALM_ADMINISTRATOR
-        is_guest = row['role'] == UserProfile.ROLE_GUEST
-        is_bot = row['is_bot']
-        # This format should align with get_cross_realm_dicts() and notify_created_user
-        result = dict(
-            email=row['email'],
-            user_id=row['id'],
-            avatar_url=avatar_url,
-            is_admin=is_admin,
-            is_guest=is_guest,
-            is_bot=is_bot,
-            full_name=row['full_name'],
-            timezone=row['timezone'],
-            is_active = row['is_active'],
-            date_joined = row['date_joined'].isoformat(),
-        )
-
-        if (realm.email_address_visibility == Realm.EMAIL_ADDRESS_VISIBILITY_ADMINS and
-                user_profile.is_realm_admin):
-            result['delivery_email'] = row['delivery_email']
-
-        if is_bot:
-            result["bot_type"] = row["bot_type"]
-            if row['email'] in settings.CROSS_REALM_BOT_EMAILS:
-                result['is_cross_realm_bot'] = True
-
-            # Note that bot_owner_id can be None with legacy data.
-            result['bot_owner_id'] = row['bot_owner_id']
-        elif include_custom_profile_fields:
-            result['profile_data'] = profiles_by_user_id.get(row['id'], {})
-        return result
-
     return {
-        row['id']: user_data(row)
+        row['id']: user_data(realm,
+                             user_profile = user_profile,
+                             row=row,
+                             client_gravatar= client_gravatar,
+                             include_custom_profile_fields=include_custom_profile_fields)
         for row in user_dicts
     }
 
+def user_data(realm: Realm, user_profile: UserProfile, row: Dict[str, Any], client_gravatar: bool,
+              include_custom_profile_fields: bool=True) -> Dict[str, Any]:
+    if include_custom_profile_fields:
```

Comment:
I think this should stay in the `raw-data` function.

Reply:
@aero31aero Thanks !

Reply:
@Hypro999 Issue 13638 was not opened at the time of making this commit .